frontend/src/app/services/game.ts

```diff
@@ -58,57 +70,93 @@ export class Game {
 
   get lastDate() {
     const lastState = last(this.simulation.modelStates);
-    return lastState?.date;
+    if (!lastState) throw new Error('Absent model state');
+
+    return lastState.date;
   }
 
   isFinished() {
-    const lastState = last(this.simulation.modelStates);
-    return Boolean(this.lastDate && this.lastDate >= this.endDate);
+    return this.lastDate >= this.endDate;
   }
 
   getMitigationState() {
     return last(this.mitigationStates);
   }
 
-  calcMitigationEffect(mitigationState: MitigationState): MitigationEffect {
+  sanitizeMitigationState(mitigationState: any) {
+    this.mitigationExclusivity.forEach(ids => {
+      let active = false;
+      ids.forEach(id => {
+        if (active) mitigationState[id].active = false;
+        active ||= mitigationState[id].active;
+      });
+    });
+  }
+
+  calcMitigationEffect(mitigationState: MitigationState, forDate: string): MitigationEffect {
     let mult = 1.0;
     let cost = 0;
     let stabilityCost = 0;
+    let bordersClosed = false;
+    const vaccinationPerDay = this.vaccinationStartDate <= forDate ? this.vaccinationPerDay : 0;
+    const monthAsString = forDate.slice(5, 7);
+    const isSchoolBreak = monthAsString === '07' || monthAsString === '08';
```

Comment:
you can `forDate.slice(5, 7) in ['07', '08']`, although date fns would be a bit more robust